src/subdomains/supporting/payment/services/fee.service.ts

```diff
@@ -134,14 +138,21 @@ export class FeeService {
 
   // --- HELPER METHODS --- //
 
-  private calculateFee(fees: Fee[], userDataId?: number): FeeDto {
+  private async calculateFee(fees: Fee[], userDataId?: number): Promise<FeeDto> {
     // get min custom fee
     const customFee = Util.minObj(
       fees.filter((fee) => fee.type === FeeType.CUSTOM),
       'rate',
     );
-    if (customFee)
-      return { rate: customFee.rate, fixed: customFee.fixed ?? 0, payoutRefBonus: customFee.payoutRefBonus };
+    if (customFee) {
+      await this.feeRepo.update(...customFee.increaseTxUsage());
+      return {
+        feeIds: customFee.id.toString(),
+        rate: customFee.rate,
+        fixed: customFee.fixed ?? 0,
+        payoutRefBonus: customFee.payoutRefBonus,
+      };
+    }
```

Comment:
I think we cannot just update the usage here. This might be called by the payment info endpoint. I think we need a different method for this